tests/Microsoft.Identity.Test.Integration.Win8/DeviceAuthenticationTests.cs

```diff
@@ -21,42 +22,46 @@ public class DeviceAuthenticationTests
+                    .WithHttpClientFactory(factory)
+                    .Build();
 
-            //Arrange
-            var labResponse = await LabUserHelper.GetSpecificUserAsync(_deviceAuthuser).ConfigureAwait(false);
-            var factory = new HttpSnifferClientFactory();
-            var msalPublicClient = PublicClientApplicationBuilder
-                .Create(labResponse.App.AppId)
-                .WithHttpClientFactory(factory)
-                .Build();
-
-            //Act
-            var authResult = msalPublicClient.AcquireTokenByUsernamePassword(new[] { "user.read" }, _deviceAuthuser, new NetworkCredential("", labResponse.User.GetOrFetchPassword()).SecurePassword)
-            .WithClaims(JObject.Parse(_claims).ToString())
-            .ExecuteAsync(CancellationToken.None).Result;
-
-            //Assert
-            Assert.IsNotNull(authResult);
-            Assert.IsNotNull(authResult.AccessToken);
-            Assert.IsNotNull(authResult.IdToken);
-            Assert.IsTrue(string.Equals(_deviceAuthuser, authResult.Account.Username, StringComparison.InvariantCultureIgnoreCase));
-
-            var (req, res) = factory.RequestsAndResponses
-                .Where(x => x.Item1.RequestUri.AbsoluteUri == labResponse.Lab.Authority + "organizations/oauth2/v2.0/token" 
-                         && x.Item2.StatusCode == HttpStatusCode.OK).ElementAt(1);
-
-            var AuthHeader = req.Headers.Single(h => h.Key == "Authorization").Value.FirstOrDefault();
-
-            Assert.IsTrue(!string.IsNullOrEmpty(AuthHeader));
-            Assert.IsTrue(AuthHeader.Contains("PKeyAuth"));
-        }
+                //Act
+               var authResult = msalPublicClient.AcquireTokenByUsernamePassword(
+                    new[] { "user.read" },
+                    labResponse.User.Upn,
+                    new NetworkCredential("", labResponse.User.GetOrFetchPassword()).SecurePassword)
+                .WithClaims(JObject.Parse(_claims).ToString())
+                .ExecuteAsync(CancellationToken.None).Result;
+
+                //Assert
+                Assert.IsNotNull(authResult);
+                Assert.IsNotNull(authResult.AccessToken);
+                Assert.IsNotNull(authResult.IdToken);
+                Assert.IsTrue(string.Equals(_deviceAuthuser, authResult.Account.Username, StringComparison.InvariantCultureIgnoreCase));
+
+                var (req, res) = factory.RequestsAndResponses
+                    .Where(x => x.Item1.RequestUri.AbsoluteUri == labResponse.Lab.Authority + "organizations/oauth2/v2.0/token"
+                             && x.Item2.StatusCode == HttpStatusCode.OK).ElementAt(1);
+
+                var AuthHeader = req.Headers.Single(h => h.Key == "Authorization").Value.FirstOrDefault();
 
+                Assert.IsTrue(!string.IsNullOrEmpty(AuthHeader));
+                Assert.IsTrue(AuthHeader.Contains("PKeyAuth"));
+            }
+            catch(Exception ex)
```

Comment:
Is this needed? Exceptions fail the tests.

Reply:
it was for testing. have removed.

---
In reply to: [529523144](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2235#discussion_r529523144) [](ancestors = 529523144)